### PR TITLE
Clean up nits

### PR DESCRIPTION
**Pull request by reyang**
There is no functionality change, just style/rename/cleanup.

## Changes
* Moved all the `AssemblyInfo.cs` file to the project root folder rather than having their own folder.
* Cleaned up the confusing term `dependencies`.
* Removed non-existing assemblies from `AssemblyInfo.cs`.

Please provide a brief description of the changes here. Update the
`CHANGELOG.md` for non-trivial changes.

For significant contributions please make sure you have completed the following items:

* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry/AssemblyInfo.cs [hunk 1]
```diff
@@ -17,7 +17,6 @@
 using System.Runtime.CompilerServices;
 
 [assembly: InternalsVisibleTo("OpenTelemetry.Tests" + AssemblyInfo.PublicKey)]
-[assembly: InternalsVisibleTo("OpenTelemetry.Instrumentation.Dependencies.Tests" + AssemblyInfo.PublicKey)]
 [assembly: InternalsVisibleTo("OpenTelemetry.Instrumentation.AspNetCore.Tests" + AssemblyInfo.PublicKey)]
 [assembly: InternalsVisibleTo("OpenTelemetry.Instrumentation.StackExchangeRedis.Tests" + AssemblyInfo.PublicKey)]
```
Comment (reyang) [inline src/OpenTelemetry/AssemblyInfo.cs:21]: @CodeBlanch is this needed? If yes, this seems like a design flaw in the SDK - the SDK should provide sufficient interface for 3rd party integration to have proper unit test, without putting downstream assemblies in its trust/visibility boundary.

@cijothomas We probably should remove `AspNetCore.Tests` from this file, too.

Comment (eddynaka) [reply src/OpenTelemetry/AssemblyInfo.cs:21]: i think we can remove. At least, locally, when i removed it didn't show any error.

Comment (reyang) [reply src/OpenTelemetry/AssemblyInfo.cs:21]: I'll keep it for now as it is beyond the scope of this PR 😄

Comment (CodeBlanch) [reply src/OpenTelemetry/AssemblyInfo.cs:21]: @reyang No idea! I checked the history, looks like it has always been in there. As long as build and tests pass, removing sounds good to me.

**Pull request merged**